public/app/plugins/datasource/tempo/traceql/TraceQLEditor.tsx

```diff
@@ -50,11 +53,40 @@ export function TraceQLEditor(props: Props) {
       onEditorDidMount={(editor, monaco) => {
         setupAutocompleteFn(editor, monaco);
         setupActions(editor, monaco, onRunQuery);
+        setupPlaceholder(editor, monaco, styles);
       }}
     />
   );
 }
 
+function setupPlaceholder(editor: monacoTypes.editor.IStandaloneCodeEditor, monaco: Monaco, styles: EditorStyles) {
+  const placeholderDecorators = [
+    {
+      range: new monaco.Range(1, 1, 1, 1),
+      options: {
+        className: styles.placeholder,
```

Comment:
Let's just add a comment here that the placeholder content is in the CSS style itself so it's easier to understand this code.

Reply:
[Done](https://github.com/grafana/grafana/pull/55257/commits/6148cf926c1302bc769e4315c9b61ae6f5ed23a6). Looks good?

Reply:
Yep that works, thanks :)